src/sentry/web/frontend/oauth_token.py

```diff
@@ -42,28 +45,46 @@ def error(self, request: HttpRequest, name, reason=None, status=400):
 
     @never_cache
     def post(self, request: HttpRequest) -> HttpResponse:
+        client_details = self._get_client_details_if_valid_request(request=request)
+
+        if not isinstance(client_details, dict):
```

Comment:
If we consolidate the response types, then instead of checking the helper response `type`, we can just check for something like 
```
if client_details.get('error_msg'):
    return self.error(request, client_details.get('error_msg'))
```

Reply:
flattened stuff out